src/bullsAndCows.js

```diff
@@ -27,6 +27,37 @@
  */
 function bullsAndCows(generatedNumber, enteredNumber) {
   // write code here
+  const result = {
+    bulls: 0,
+    cows: 0,
+  };
+
+  const regex = /[A-Za-z]/g;
+
+  if (enteredNumber === undefined || enteredNumber.length !== 4
+    || enteredNumber.match(regex) !== null) {
+    return undefined;
+  };
+
+  for (let i = 0; i < generatedNumber.length; i++) {
+    if (enteredNumber.indexOf(enteredNumber[i])
+    !== enteredNumber.lastIndexOf(enteredNumber[i])) {
+      return undefined;
+    }
+
+    if (generatedNumber[i] === enteredNumber[i]) {
+      result.bulls++;
+    }
+  }
+
+  for (let j = 0; j < generatedNumber.length; j++) {
+    if (generatedNumber.indexOf(enteredNumber[j]) !== -1
+      && generatedNumber[j] !== enteredNumber[j]) {
+      result.cows++;
+    }
+  }
```

Comment:
Второй цикл не нужен
```suggestion
    if (generatedNumber[i] === enteredNumber[i]) {
      result.bulls++;
    } else ... 
    result.cows++
```

Reply:
👌